feat/disponibilidadePersonal

VitorSuave opened the pull request:
aaaaa

VitorSuave committed:
feat/(disponibilidadePersonal): Foi criado todas as classes referente ao CRUD de criação e manipulção para o cadastro dos horarios. Assim como as validações de sobreposição e calculo da disponibilidade, foi alterado também a tipagem do repository do personal com a entendidade do usuario e padronizando para Long

# Conflicts:
#	.gitignore
#	src/main/java/com/spring/ApiSystem/agendamento/AgendamentoService.java
#	src/main/java/com/spring/ApiSystem/agendamento/mapper/AgendamentoMapper.java
#	src/main/java/com/spring/ApiSystem/personal/PersonalController.java
#	src/main/java/com/spring/ApiSystem/personal/PersonalService.java

VitorSuave committed:
feat/(disponibilidadePersonal): Foi criado todas as classes referente ao CRUD de criação e manipulção para o cadastro dos horarios. Assim como as validações de sobreposição e calculo da disponibilidade, foi alterado também a tipagem do repository do personal com a entendidade do usuario e padronizando para Long

VitorSuave committed:
feat(disponibilidadePersonal):Realizado um select no banco para retornar os dias disponiveis e os horarios do personal. Assim a gente mentem menos chamadas ao banco.

VitorSuave committed:
feat(disponibilidadePersonal):Realizado um select no banco para retornar os dias disponiveis e os horarios do personal. Assim a gente mentem menos chamadas ao banco.

VitorSuave committed:
fix(Modificação do metodo de buscar horarios dispo): substitui métodos auxiliares por lógica única em obterDisponibilidadePersonal - Gera slots de 15min a partir de data completa - Exclui horários de tipos INTERVALO e RESTRITO - Simplifica fluxo de obtenção de disponibilidade

VitorSuave committed:
fix(Modificação do metodo de buscar horarios dispo): substitui métodos auxiliares por lógica única em obterDisponibilidadePersonal - Gera slots de 15min a partir de data completa - Exclui horários de tipos INTERVALO e RESTRITO - Simplifica fluxo de obtenção de disponibilidade

VitorSuave committed:
Merge remote-tracking branch 'origin/develop' into feat/disponibilidadePersonal

# Conflicts:
#	src/main/java/com/spring/ApiSystem/agendamento/AgendamentoService.java
#	src/main/java/com/spring/ApiSystem/personal/PersonalService.java

VitorSuave committed:
fix(exceeptions): Alteração das mensagem dos exception

VitorSuave committed:
fix(exceeptions): Alteração das mensagem dos exception

JoaoGaldinoCunha commented:
irei olhar

JoaoGaldinoCunha commented on src/main/java/com/spring/ApiSystem/horariopersonal/DisponibilidadePersonalService.java:
```diff
@@ -0,0 +1,206 @@
+
+
+    // Consulta as disponibilidades dos horarios de um personal.
+    @Transactional(readOnly = true)
```
ajustar conforme alinhado

VitorSuave replied:
@JoaoGaldinoCunha Quando puder dar uma olhada me avisa, entre hoje mais tarde e amanha eu vou testar mais coisas.

JoaoGaldinoCunha reviewed (changes requested):
realizar ajustes

VitorSuave committed:
fix: Atualização da lógica de disponibilidade e Buffer(Intervalo) entre agendamentos.

Melhorias e Ajustes: - Validação de Conflito: Regras implementadas para evitar sobreposição entre Restrições e Agendamentos Ativos + Intervalo Pós-Atendimento. - Buffer Dinâmico: Implementação do Intervalo Sessões (15-60 min) com aplicação do buffer APENAS após o atendimento. - Regras de Restrição: Adicionado os 15 minutos ANTES do início de blocos RESTRITO(Almoço). - Tempo Hábil: Lógica no GET para filtrar slots que não permitem a duração mínima da aula (30 min). - Endpoint: Refatorado PUT /horarios/{horarioId} para sobrescrita de blocos individuais. - Inicialização: Adicionado criação de horários default no cadastro do Personal.

VitorSuave committed:
fix: Atualização da lógica de disponibilidade e Buffer(Intervalo) entre agendamentos.

Melhorias e Ajustes: - Validação de Conflito: Regras implementadas para evitar sobreposição entre Restrições e Agendamentos Ativos + Intervalo Pós-Atendimento. - Buffer Dinâmico: Implementação do Intervalo Sessões (15-60 min) com aplicação do buffer APENAS após o atendimento. - Regras de Restrição: Adicionado os 15 minutos ANTES do início de blocos RESTRITO(Almoço). - Tempo Hábil: Lógica no GET para filtrar slots que não permitem a duração mínima da aula (30 min). - Endpoint: Refatorado PUT /horarios/{horarioId} para sobrescrita de blocos individuais. - Inicialização: Adicionado criação de horários default no cadastro do Personal.

VitorSuave committed:
fix: ASó atualizando algumas informações referente aos comenÃtários.

VitorSuave committed:
fix: ASó atualizando algumas informações referente aos comenÃtários.

JoaoGaldinoCunha reviewed (commented):
Tudo Ok, mas da um rebase da Develop para ajustar diversos conflitos que vão acontecer, e se tiver testes teste primeiro depois valido!

VitorSuave committed:
atualizando o endpoint para /personais

VitorSuave committed:
atualizando o endpoint para /personais

VitorSuave committed:
fix: Implementado atualização do metodo atualizarBufferMinutos para mexer no intervalo entre sessões do personal. Foi criado um DTO para isso.

VitorSuave committed:
fix: Implementado atualização do metodo atualizarBufferMinutos para mexer no intervalo entre sessões do personal. Foi criado um DTO para isso.

VitorSuave committed:
fix: Apagando metodos que não serão mais usados

VitorSuave committed:
fix: Apagando metodos que não serão mais usados

VitorSuave committed:
fix: bugs

VitorSuave committed:
Merge - feat/disponibilidadePersonal to develop

JoaoGaldinoCunha commented:
@VitorSuave ta pronto?

VitorSuave commented:
> @VitorSuave ta pronto?
Sim, to resolvendo os conflitos que deu agora depois do rebase e merge, referente as mudanças do agendamento. Dps confere teu whats te mandei sobre.

VitorSuave committed:
fix(ajustes no select sobre os agendamentos): Foi feito alguns ajustes.

VitorSuave committed:
adicionado testes

JoaoGaldinoCunha reviewed (approved):
ACEITOOOO

VitorSuave committed:
Merge branch 'develop' into feat/disponibilidadePersonal
